python/pyspark/sql/catalog.py

```diff
@@ -29,31 +31,35 @@
 Column = namedtuple("Column", "name description dataType nullable isPartition isBucket")
 Function = namedtuple("Function", "name description className isTemporary")
 
+if TYPE_CHECKING:
+    from pyspark.sql._typing import UserDefinedFunctionLike
+    from pyspark.sql.types import DataType
+
 
 class Catalog(object):
     """User-facing catalog API, accessible through `SparkSession.catalog`.
 
     This is a thin wrapper around its Scala implementation org.apache.spark.sql.catalog.Catalog.
     """
 
-    def __init__(self, sparkSession):
+    def __init__(self, sparkSession: SparkSession) -> None:
         """Create a new Catalog that wraps the underlying JVM object."""
         self._sparkSession = sparkSession
-        self._jsparkSession = sparkSession._jsparkSession
-        self._jcatalog = sparkSession._jsparkSession.catalog()
+        self._jsparkSession = sparkSession._jsparkSession  # type: ignore[attr-defined]
+        self._jcatalog = sparkSession._jsparkSession.catalog()  # type: ignore[attr-defined]
 
     @since(2.0)
-    def currentDatabase(self):
+    def currentDatabase(self) -> str:
         """Returns the current default database in this session."""
         return self._jcatalog.currentDatabase()
 
     @since(2.0)
-    def setCurrentDatabase(self, dbName):
+    def setCurrentDatabase(self, dbName: str) -> None:
         """Sets the current default database in this session."""
         return self._jcatalog.setCurrentDatabase(dbName)
```

Comment:
Can we fix this as well? We should have no `return` if function is not expected to return anything.

Reply:
Good catch! Would you mind if fixing that in https://github.com/apache/spark/pull/34147? There are two more APIs to be fixed similarly.

Reply:
Would you help confirm if we may remove the `return`  here since there are no user-facing changes introduced? CC @HyukjinKwon @ueshin

Reply:
I think it's fine to remove `return` here.